src/btreemap.rs

```diff
@@ -1161,156 +1161,317 @@ mod test {
         VectorMemory,
     };
     use std::cell::RefCell;
+    use std::convert::TryFrom;
     use std::rc::Rc;
 
-    pub(crate) fn make_memory() -> Rc<RefCell<Vec<u8>>> {
-        Rc::new(RefCell::new(Vec::new()))
+    /// Returns a fixed‑size buffer for the given u32.
+    fn monotonic_buffer<const N: usize>(i: u32) -> [u8; N] {
+        let mut buf = [0u8; N];
+        let bytes = i.to_be_bytes();
+        buf[N - bytes.len()..].copy_from_slice(&bytes);
+        buf
+    }
+
+    #[test]
+    fn test_monotonic_buffer() {
+        let cases: &[(u32, [u8; 4])] = &[
+            (1, [0, 0, 0, 1]),
+            (2, [0, 0, 0, 2]),
+            ((1 << 8) - 1, [0, 0, 0, 255]),
+            ((1 << 8), [0, 0, 1, 0]),
+            ((1 << 16) - 1, [0, 0, 255, 255]),
+            (1 << 16, [0, 1, 0, 0]),
+            ((1 << 24) - 1, [0, 255, 255, 255]),
+            (1 << 24, [1, 0, 0, 0]),
+        ];
+
+        for &(input, expected) in cases {
+            let output = monotonic_buffer::<4>(input);
+            assert_eq!(
+                output, expected,
+                "monotonic_buffer::<4>({}) returned {:?}, expected {:?}",
+                input, output, expected
+            );
+        }
+    }
+
+    /// A trait to construct a value from a u32.
+    trait Builder {
```

Comment:
FYI: this trait provides a type abstraction and improves readability of each test.

In order not to write `K::build(i)`/`V::build(i)` I prefer to add a `key`/`value` closures to each test to write `key(i)`/`value(i)`.